src/ArmTemplates/Commands/Executors/ExtractorExecutor.cs

```diff
@@ -459,6 +460,81 @@ public async Task<Template<GroupTemplateResources>> GenerateGroupsTemplateAsync(
             return templateParameters;
         }
 
+        public async Task GenerateResourceParametersFiles(
```

Comment:
Do we really need a top level function here? What about generating and storing the parameters file for dedicated resource in it's special method?
Like
```
public async Task<Template<ApiTemplateResources>> GenerateApiTemplateAsync(
            string singleApiName,
            List<string> multipleApiNames,
            string baseFilesGenerationDirectory)
        {
            // normal extractor behaviour here
            
            var parametersFile = await this.parametersGenerator.GenerateParameters<ApiTemplateResources>(apiTemplates);
            
            if (parametersFile ...)
            {
                ....
            }
            
            this.logger.LogInformation("Finished generation of api template...");
            return apiTemplate;
        }
```

Reply:
I was thinking to go this way first, but here are some thoughts: 

1. In order to generate main parameters file we have to have resources already extracted (for example identityProviders, OpenIdConnectProviders)
2. The way introduced here: we are relying on only one method of generation of parameters (GenerateParametersTemplateAsync) from which all other parameters are generated. So, generation of parameters executes only ones, and the method just collects needed parameters without logic behind. 
3. In the suggested way we have to generate parameters for specific resource plus include the same logic into generation of main parameter template.